Fix global component loading in production builds.

rillian opened the pull request:
Instead of maintaining an index.js which re-exports all the vue.js modules in the global-components directory, use webpack's `require.context` loader to iterate over all the contained files and register the components from there.

The previous method constructed component names from the `__file` property of the module objects, but this is only available in debug builds and so the site failed to load when deployed.

Closes #32. Based on [this example](https://vuejs.org/v2/guide/components-registration.html#Automatic-Global-Registration-of-Base-Components) from the vue.js guild.

rillian committed:
Fix global component loading in production builds.

Instead of maintaining an index.js which re-exports all the
vue.js modules in the global-components directory, use
webpack's `require.context` loader to iterate over all
the contained files and register the components from there.

The previous method constructed the component names from
the __file property of the module objects, but this is only
available in debug builds and so the site failed to load
when deployed.

rillian commented:
Looks like the [test deployment](https://5cfeb30001ba6500093da403--readhomer-dev.netlify.com/) works.

jacobwegner commented:
Thanks for the PR, @rillian.  I'm pulling @paltman in here for a review.

@paltman, it looks like we had a regression in b393184d9a11fd0463223f19782860ef671c56e8  after my previous fix for this issue in 16f106527bef04893ecfbd94b64808f5fdd323e1.

I like @rillian's implementation better.  I think the only other thing we might want to keep in mind is ensuring that each module within `global-components` always defines a `default` export, but I like the idea of dropping the explicit registry in `global-components/index.js`.

paltman commented:
I'd prefer the solution in 16f106527bef04893ecfbd94b64808f5fdd323e1 so we can be explicit about what's loading at out of that directory.  But also know that the latest skeleton doesn't even have the notion of global-components anymore.

rillian committed:
Restore explicit loading of global components.

Patrick Altman prefers an explicit export list rather than
loading everything in the directory.

Restore global-components/index.js with an explicit list of
default exports, which was removed in the previous commit,
and revert to Jacob Wegner's loading solution from commit
16f106527bef04893ecfbd94b64808f5fdd323e1.

Instead of deriving the component name from the debug-only
__file property, export the desired components as properties
of the default object and use Object.entries to enumerate over
them, using the property name, and hence module import name, as the
registered component name.

rillian commented:
Thanks for commenting. I've restored @jacobwegner's solution with the explicit index.js.

> But also know that the latest skeleton doesn't even have the notion of global-components anymore.

It would still be helpful to have a master branch which is deployable.

jacobwegner commented:
(Pulling this in so we have something working deployed to Netlify)

rillian commented:
Thanks for merging!
